src/base/async/int.c

```diff
@@ -2031,10 +2031,13 @@ static int do_redirect(int old_only)
     sda_hi = SREG(ds);
     sda_size = LWORD(ecx);
 
-    ds_printf("INT21: lol = 0x%04x\n", (lol_hi << 4) + lol_lo);
-    ds_printf("INT21: sda = 0x%04x, size = 0x%04x\n",
-	      (sda_hi << 4) + sda_lo, sda_size);
+    ds_printf("INT21: lol = %04x:%04x\n", lol_hi, lol_lo);
+    ds_printf("INT21: sda = %04x:%04x, size = 0x%04x\n", sda_hi, sda_lo, sda_size);
     ds_printf("INT21: ver = 0x%02x, 0x%02x\n", major, minor);
+    if (lol_hi != sda_hi) {
+      error("INT21: WARNING lol and sda segments differ\n");
+      leavedos(58);
```

Comment:
Oh, why so rough?
Just return failure?

Reply:
It's pretty serious if the LOL and SDA have different segments (I don't believe it could ever happen) as later we'll set them both to DOS_DATA_SEGMENT with the original offsets. An alternative would be to disable the redirector, is that better?

Reply:
Yes, much better.
We do support DOSes w/o redirector, so why to crash.

Reply:
Okay will disable the redirector instead, seems nicer.